pkg/controller/lifecycle/actuator.go

```diff
@@ -155,6 +157,7 @@ func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, ex *extension
 		extensions.GenericTokenKubeconfigSecretNameFromCluster(cluster),
 		oidcShootAccessSecret.Secret.Name,
 		generatedSecrets[constants.WebhookTLSSecretName].Name,
+		*cluster.Seed.Status.KubernetesVersion,
```

Comment:
I am not sure how reliable is to read the seed's k8s version from the `cluster.seed.status` section. Instead we can inject a clientset in the actuator object and use the `ServerVersion()` function (or something similar) to determine the k8s version. See [this](https://github.com/gardener/gardener/blob/1da39002ccd90e8b3dc08763dcf41708e48d907f/extensions/pkg/controller/worker/genericactuator/actuator.go#L108) example.